src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -216,6 +206,46 @@ public synchronized RcaResponse readRca(String rca) {
     return response;
   }
 
+  private RcaResponse readTemperatureProfileRca(String rca) {
+    RcaResponse response = null;
+    Field<Integer> primaryKeyField = DSL.field(
+        SQLiteQueryUtils.getPrimaryKeyColumnName(ResourceFlowUnit.RCA_TABLE_NAME), Integer.class);
+    SelectJoinStep<Record> rcaQuery = SQLiteQueryUtils.buildRcaQuery(create, rca);
+    try {
+      List<Record> recordList = rcaQuery.fetch();
+      if (recordList == null || recordList.isEmpty()) {
+        return null;
+      }
+      Record mostRecentRecord = recordList.get(0);
+      response = RcaResponse.buildResponse(mostRecentRecord);
+
+      if (rca.equals(ClusterTemperatureRca.TABLE_NAME)) {
```

Comment:
Can we make sure that this is the elected master we are making the query to ?

Reply:
This is taken care of by the RcaRequestHandler and the fact that cluster temperature rca is not a part of the rcas that can be queried with local=true param and is a part of cluster level rcas. Because it's not part of the local rca set, it by default falls back to ensuring that this is a cluster level rca and that flow checks if the current host is a master node already.

Reply:
Sounds good. Can we add that as a code comment ? So that people know why they cannot get this RCA when queried from non-elected masters ?

Reply:
done.